nexus/db-queries/src/db/datastore/instance.rs

```diff
@@ -1128,48 +1146,9 @@ impl DataStore {
         use db::schema::disk::dsl as disk_dsl;
         use db::schema::instance::dsl as instance_dsl;
 
-        // * Allow setting the boot disk in NoVmm because there is no VMM to
-        //   contend with.
-        // * Allow setting the boot disk in Failed to allow changing the boot
-        //   disk of a failed instance and free its boot disk for detach.
-        // * Allow setting the boot disk in Creating because one of the last
-        //   steps of instance creation, while the instance is still in
-        //   Creating, is to reconfigure the instance to the desired boot disk.
```

Comment:
maybe worth keeping a comment similar to this one, but moving it to the `filter(instance_dsl::state.eq_any(NOT_INCARNATED_STATES))` line? not a big deal though --- I do think it's nice to document someplace in here "we can't set the boot disk if the instance is incarnated because, you know..."

Reply:
took this as a nudge to actually document the property we're trying to retain on the function itself (and same for resize), thanks! i realized when re-reading this code that we've _talked_ about it but i don't think that landed anywhere closer to the code itself: "ensuring the database does not drift from actual VMMs if VMMs exist anywhere"

if we didn't care about the database being reflective of the current universe we could definitely update boot disks whenever... we just.. don't want the database to tell mistruths.